scripts/profile_model.sh

```diff
@@ -1,44 +1,40 @@
 #!/usr/bin/env bash
-set -x
+
+set -xe
+
+function _cleanup() {
+    [[ ${#DIRSTACK[@]} -gt 1 ]] && popd
+    [[ -d ${REF_TMP_DIR} ]] && rm -rf ${REF_TMP_DIR}
+}
+
+trap _cleanup EXIT
+
+# Reference is with respect to the `dev` branch, by default.
+REF_HEAD=${1:-dev}
+REF_COMMIT=$(git rev-parse --short HEAD)
+BENCHMARK_FILE=tests/perf/test_benchmark.py
+BENCHMARK_DIR=$( cd $(dirname "$0")/../.benchmarks ; pwd -P )
+REF_TMP_DIR=.tmp_test_dir
 
 CURRENT_HEAD=$(git rev-parse --abbrev-ref HEAD)
 CURRENT_COMMIT=$(git rev-parse --short HEAD)
 # If the current state is detached head, store current commit info instead.
 if [ ${CURRENT_HEAD} = 'HEAD' ]; then
-    git checkout -b tmp
     CURRENT_HEAD=$(git rev-parse HEAD)
 fi
-# Reference is with respect to the `dev` branch.
-REF_HEAD=${1:-dev}
-REF_COMMIT=$(git rev-parse --short HEAD)
-BENCHMARK_FILE=tests/perf/test_benchmark.py
-IS_BENCHMARK_FILE_IN_DEV=1
-REF_EXIT_CODE=0
-IS_DIRTY=0
 
-# If uncommitted changes exist, stash these and set a flag.
-git diff-index --quiet HEAD -- || IS_DIRTY=1
-git stash
-git checkout ${REF_HEAD}
+# clone the repo into the temporary directory and run benchmark tests
+git clone -b ${REF_HEAD} --single-branch https://github.com/uber/pyro.git ${REF_TMP_DIR}
```

Comment:
git local clones are super quick. Could this be changed to use a local clone?
```sh
git clone -b ${REF_HEAD} --single-branch . ${REF_TMP_DIR}
```

Reply:
Good idea. Let me modify this for travis vs. local runs a bit.